### PR TITLE
Feature/7720 my store summary screen

### LINKED ISSUE #7720: Domain Selection in Store Creation Flow
Add new domain selection step following design from HyVloP5FipZzyPVenH2euI-fi-1579%3A83494

Screenshot 2022-11-02 at 11 00 15

### UI
- [x] Choose domain screen -> #7732
- [x] My store final screen -> #7802
- [x] Error handling -> #7879

### Network

- [x] Fetch domain suggestions -> #7761

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #7720

### Description
Adds My store summary screen UI. This screen is presented right after domain picker step, whether by clicking "Continue" or by clicking "Skip" on domain picker step. 

### Testing instructions
Check that the design matches Figma designs HyVloP5FipZzyPVenH2euI-fi-1980%3A94456

### Images/gif

| Light  | Dark |
| ------------- | ------------- |
|   |   |

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/NewStore.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.woocommerce.android.ui.login.storecreation
+
+import kotlinx.coroutines.flow.MutableStateFlow
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class NewStore @Inject constructor() {
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/NewStore.kt:8]: @0nko I added this new class as data holder for store creation input data. I've exposed it as a stateflow as it works pretty well with the different viewmodels we are creating for the flow. 

Let me know what you think about it. Furthermore, I'm not super happy with the naming so feel free to suggest and alternative.

**Review by 0nko — COMMENTED** [review]
The screen looks good, I just noticed the Figma design contains the flag (though I'm not sure if it's been updated - the emoji was supposed to be removed based on the comment below). Also, the button title is slightly different.

image

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/NewStore.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/NewStore.kt:8]: I can't think of a name either.. Let's keep it for now and we can change it later 🤷

**Comment by JorgeMucientes** [comment]
Thanks for the review and the good catches @0nko 

- I've fixed the copy for the continue button following Figma designs
- Regarding the emoji. Currently, we don't have emojis for countries. Until the API of the profiler step with the list of countries and their emojis is ready, we won't display an emoji. In fact, for M2, the country, and the category fields won't be displayed in this screen as they won't be available. 
- I've added a few fixes in navigation to work well with the recently merged Plans screen

Ready for another round :)

**Review by 0nko — APPROVED** [review]
Looks great! 👍 Thanks for the updates.

**Pull request merged**